src/common/io.cc

```diff
@@ -388,8 +401,8 @@ int64_t read_vector(int fd, int64_t *type, std::vector<uint8_t> &buffer) {
 
 void write_log_message(int fd, const char *message) {
   /* Account for the \0 at the end of the string. */
-  write_message(fd, static_cast<int64_t>(CommonMessageType::LOG_MESSAGE),
```

Comment:
doesn't this need to be protected by a lock too?

Reply:
No, because this function isn't used by local scheduler client.
(Actually, I searched the code, it's only used in tests.)

Reply:
Yeah, I think you are right!